vllm/model_executor/layers/fused_moe.py

```diff
@@ -0,0 +1,286 @@
+"""Fused MoE kernel."""
+import torch
+import triton
+import triton.language as tl
+from vllm._C import ops
+
+
+@triton.jit
+def fused_moe_kernel(
+    # Pointers to matrices
+    a_ptr,
+    b_ptr,
+    c_ptr,
+    topk_weights_ptr,
+    sorted_token_ids_ptr,
+    expert_ids_ptr,
+    num_tokens_post_padded_ptr,
+    # Matrix dimensions
+    N,
+    K,
+    EM,
+    num_valid_tokens,
+    # The stride variables represent how much to increase the ptr by when moving by 1
+    # element in a particular dimension. E.g. `stride_am` is how much to increase `a_ptr`
+    # by to get the element one row down (A has M rows).
+    stride_am,
+    stride_ak,
+    stride_be,
+    stride_bk,
+    stride_bn,
+    stride_cm,
+    stride_cn,
+    stride_weight,
+    stride_token_id,
```

Comment:
Btw, at the moment you are not using this in the code. Instead yo assume the stride is 1 in the code below (which is a valid assumption since sorted_token_ids is allocated by moe_align_block_size). It might be clearer to remove this parameter and instead add an assert that the stride is 1.

Reply:
it's right, in this version, the stride is always 1, i have fix it and instead add a assert before invoking the kernel